src/shipchain_common/authentication.py

```diff
@@ -37,10 +37,10 @@ def get_jwt_from_request(request):
     """
     This is for retrieving the decoded JWT from the a request via the simplejwt authenticator.
     """
-    if not settings.PROFILES_ENABLED:
-        return None
-    return (request.authenticators[-1].get_raw_token(request.authenticators[-1].get_header(request)).decode()
-            if request.authenticators else None)
+    if settings.PROFILES_ENABLED and request.user.is_authenticated:
```

Comment:
Let's copy all of the logic from IsAuthenticated.has_permission here - `request.user and request.user.is_authenticated`. I think request.user is present in most cases, but it's a good check to have in place.